src/Zip.hs

```diff
@@ -64,26 +70,30 @@ writeZips BucketFiles{..} =
 
 writeZip :: (MonadThrow m, MonadResource m) => Text -> m ()
 writeZip pathToFile = do
-  let stringPath = unpack pathToFile
-  let nameOfZip = (takeFileName . dropTrailingPathSeparator $ stringPath) <> ".zip"
+  let stringPath = dropTrailingPathSeparator . unpack $ pathToFile
+  let nameOfZip = takeFileName stringPath <> ".zip"
   void . register $ deleteFileIfExists nameOfZip
   (liftIO . doesDirectoryExist) stringPath >>= \case
-    True -> liftIO . writeFolderToZip stringPath $ nameOfZip
+    True -> liftIO $
```

Comment:
You can do the same rearrangement of the pattern match here with ViewPatterns/PatternGuards that I referenced in an earlier comment.